engine/src/main/java/org/terasology/entitySystem/entity/internal/EngineEntityPool.java

```diff
@@ -69,4 +71,20 @@
      * @return whether the entity has the component
      */
     boolean hasComponent(long entityId, Class<? extends Component> componentClass);
+
+    /**
+     * Remove the entity from the pool. This does not destroy the entity, it only removes the {@link EntityRef}
+     * and the {@link Component}s from this pool, so that the entity can be moved to a different pool. It does
+     * not invalidate the {@link EntityRef}.
+     *
+     * Returns an {@link Optional} {@link BaseEntityRef} if it was removed, ready to be put into another pool. If nothing
+     * was removed, return {@link Optional#empty()}.
+     *
+     * @param id the id of the entity to remove
+     * @return an optional {@link BaseEntityRef}, containing the removed entity
+     */
+    Optional<BaseEntityRef> remove(long id);
```

Comment:
This is internal API that needs to be handled with care as I understand it. How big is the potential risk for leaking entity refs here, i.e., remove them from the pool but don't put them anywhere else?